corehq/form_processor/tests/test_basics.py

```diff
@@ -500,6 +503,60 @@ def test_caching_form_attachment_during_submission(self):
         with patch.object(get_blob_db(), 'get', side_effect=Exception('unexpected blobdb read')):
             _submit_case_block(True, uuid.uuid4().hex, user_id='user2', update={})
 
+    def test_failed_form_with_case_with_secret(self):
+        with override_settings(XFORM_PRE_PROCESSORS={
+            DOMAIN: ['corehq.apps.data_vault.tests.test_utils.TestVaultPatternExtractor']}
+        ):
+            case_id = uuid.uuid4().hex
+            modified_on = datetime.utcnow()
+            self.assertEqual(VaultEntry.objects.count(), 0)
+
+            _submit_case_block(
+                True, case_id, user_id='user1', owner_id='owner1', case_type='demo',
+                case_name='this is a very long case name that exceeds the 255 char limit' * 5,
+                date_modified=modified_on, date_opened=modified_on, update={
+                    'dynamic': '123', 'secret_case_property': '7777777777'
+                }
+            )
+            with self.assertRaises(CaseNotFound):
+                self.casedb.get_case(case_id)
```

Comment:
should this also validate that the error form is saved and doesn't contain the secret?

Reply:
that is done when testing form submission in `corehq/apps/receiverwrapper/tests/test_submissions.py`

Reply:
nvm, i will try to cover that here as well, worth testing.